src/YoutubeAtomPlayer.tsx

```diff
@@ -0,0 +1,276 @@
+import React, { useEffect, useRef } from 'react';
+import YouTubePlayer from 'youtube-player';
+
+import type { AdTargeting, ImageSource, VideoEventKey } from './types';
+import type { ConsentState } from '@guardian/consent-management-platform/dist/types';
+import {
+    AdsConfig,
+    buildAdsConfigWithConsent,
+    disabledAds,
+} from '@guardian/commercial-core';
+import { log } from '@guardian/libs';
+
+type Props = {
+    assetId: string;
+    overrideImage?: ImageSource[];
+    posterImage?: ImageSource[];
+    adTargeting?: AdTargeting;
+    consentState?: ConsentState;
+    height: number;
+    width: number;
+    title?: string;
+    origin?: string;
+    eventEmitters: ((event: VideoEventKey) => void)[];
+    autoPlay: boolean;
+    loadPlayer: boolean;
```

Comment:
Similar to my other content, we could raise `loadPlayer` and `consentState` up and use them as conditionals? The player component then becomes simpler.

But again, happy to take your lead here and I also see the other side to this argument where it might be desirable to make it difficult to ever load the player without consent

Reply:
Good idea thanks - it makes the player less semantically coupled.

I've lifted `loadPlayer` and `consentState` up and created an `onReady` callback.

This also allows us to make `consentState` mandatory to `YoutubeAtomPlayer`.

The only drawback is because the callback is a dependency in an `useEffect`, I had to create the callback with `useCallback` so it is stable across renders.